src/binding/transaction_log_file_windows.cpp

```diff
@@ -73,8 +73,101 @@ void TransactionLogFile::openFile() {
 		throw std::runtime_error("Failed to get file size: " + this->path.string());
 	}
 	this->size = static_cast<size_t>(fileSize.QuadPart);
+	DEBUG_LOG(stderr, "File size: %zu file path: %s\n", this->size, this->path.string().c_str());
+	// On Windows, we have to create the full file size for memory maps, and it is zero-padded, so the act of indexing allows us to find
+	// the end, and adjust the real size accordingly.
+	// TODO: Future optimization is to only do this if the file is a multiple of the page size, and ensure
+	// files that are expanded to a memory page are memory page aligned, with (this->size & 0xFFF) == 0
+	if (this->size > 0) {
+		this->findPositionByTimestamp(0, this->size);
+		DEBUG_LOG("New file size: %zu file path: %s\n", this->size, this->path.string().c_str());
+	}
 }
 
+MemoryMap* TransactionLogFile::getMemoryMap(uint32_t fileSize) {
+	DEBUG_LOG("%p TransactionLogFile::getMemoryMap open size: %u\n", this, fileSize);
+	if (memoryMap) {
+		if (memoryMap->mapSize >= fileSize) {
+			// existing memory map will work
+			memoryMap->fileSize = fileSize;
+			return memoryMap;
+		}
+		DEBUG_LOG("%p TransactionLogFile::getMemoryMap existing memory map was too small: %u\n", this, memoryMap->mapSize);
+		// this memory map is not big enough, need to create a new one
+		if (--memoryMap->refCount == 0) { // unref it
+			delete memoryMap;
+		}
+	}
+	DEBUG_LOG("%p TransactionLogFile::getMemoryMap creating new memory map: %u\n", this, fileSize);
+	// In windows, we can not map beyond the size of the file (without using driver-level APIs that directly call procedures
+	// in NT.DLL). So we must expand the file to the full size before we can map it.
+	if (fileSize > this->size)
+	{
+		LARGE_INTEGER currentPos;
+		LARGE_INTEGER distanceToMove;
+		// First, we have to get the current position, so we can restore it (if we get to a point where no other code relies on position, could remove this)
+		distanceToMove.QuadPart = 0; // We want to move 0 bytes to query current position
+		if (!SetFilePointerEx(this->fileHandle, distanceToMove, &currentPos, FILE_CURRENT)) {
```

Comment:
While not required, it's a good habit to specify the scope resolution operator to all external APIs.

```suggestion
		if (!::SetFilePointerEx(this->fileHandle, distanceToMove, &currentPos, FILE_CURRENT)) {
```